BioinformaticTools-inator/modules/dna_rna_funcs.py

```diff
@@ -0,0 +1,110 @@
+from typing import Dict, List, Union
+
+dna_dict = {
+    'A': 'T', 'G': 'C', 'T': 'A', 'C': 'G',
+    'a': 't', 'g': 'c', 't': 'a', 'c': 'g'
+    }
+rna_dict = {
+    'A': 'U', 'G': 'C', 'U': 'A', 'C': 'G',
+    'a': 'u', 'g': 'c', 'u': 'a', 'c': 'g'
+    }
+
+
+def is_valid_na(seq: str) -> None:
+    if ('u' in seq or 'U' in seq) and ('t' in seq or 'T' in seq):
+        raise ValueError("Error: simultaneous presence of 'u' and 't' "
+                         " in any register is unacceptable.")
+    if not set([symb.upper() for symb in seq]).issubset(
+            {'A', 'T', 'G', 'C', 'U'}
+            ):
+        raise ValueError(f"Error: the sequence contains "
+                         f"invalid characters: {seq}")
+
+
+def transcribe(*args: str) -> Union[List[str], str]:
+    results = []
+    for seq in args:
+        transcribed_seq = seq.replace('T', 'U').replace('t', 'u')
+        results.append(transcribed_seq)
+    if len(args) == 1:
+        return results[0]
+    return results
+
+
+def reverse(*args: str) -> Union[List[str], str]:
+    results = [seq[::-1] for seq in args]
+    if len(args) == 1:
+        return results[0]
+    return results

```

Comment:
Кажется: что можно возвращать results в main, а уже там один раз прописать, вот это условие. Плюс такого подхода в том, что при желании изменить эту часть кода, тебе не придется искать каждую желаемую строку, потому что она будет у тебя в единственном экземпляре 
```suggestion

    return results

```

Reply:
Если вспомнить мое предложение по исправлению в мэйне вот это 
```
    if operation in fn_map.keys():
        fn = dna_rna_funcs.fn_map[operation] 
        return fn(*seqs)
```
то теперь оно превращается в 
```
if operation in fn_map.keys():
        fn = dna_rna_funcs.fn_map[operation] 
if len(result) == 1:
        return result[0]
return result
```
Больше я это исправление не буду прописывать, но оно актуально для всех конструкций, про которое я говорила чуть выше